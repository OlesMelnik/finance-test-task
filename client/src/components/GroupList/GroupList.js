import React, { useState } from "react";
import AddIcon from "@mui/icons-material/Add";
import DeleteIcon from "@mui/icons-material/Delete";
import Stack from "@mui/material/Stack";
import Button from "@mui/material/Button";
import { GroupDialogs } from "../GroupDialogs";
import { GroupBadge } from "../GroupBadge";
import { ConfirmModal } from "../ConfirmModal";

export const GroupList = ({
  groups,
  tickers,
  onSelect,
  onDelete,
  selectedGroup,
}) => {
  const [showDialog, setShowDialog] = useState(false);
  const [showConfirmModal, setShowConfirmModal] = useState(false);
  const openConfirmModal = () => setShowConfirmModal(true);
  const hideConfirmModal = () => setShowConfirmModal(false);
  const openDialog = () => setShowDialog(true);
  const closeDialog = () => setShowDialog(false);

  return (
    <Stack direction="row" spacing={2}>
      <GroupBadge
        onSelect={() => onSelect("All")}
        group={{ name: "All", tickers: tickers }}
        selectedGroup={!selectedGroup && { name: "All" }}
      />
      {groups &&
        !!groups.length &&
        groups.map((group) => (
          <GroupBadge
            onSelect={() => onSelect(group.name)}
            group={group}
            selectedGroup={selectedGroup}
          />
        ))}
      <Button
        startIcon={<AddIcon />}
        style={{ textTransform: "none" }}
        onClick={openDialog}
      >
        Новий список
      </Button>
      {selectedGroup && (
        <>
          <Button
            startIcon={<DeleteIcon />}
            style={{ textTransform: "none" }}
            onClick={openConfirmModal}
            color="error"
          >
            Видалити список
          </Button>
          <ConfirmModal
            message={"Ви впевнені, що хочети видалити цю групу?"}
            open={showConfirmModal}
            onConfirm={() => {
              onDelete();
              hideConfirmModal();
            }}
            onCancel={hideConfirmModal}
          />
        </>
      )}
      <GroupDialogs
        showFirstDialog={showDialog}
        closeFirstDialog={closeDialog}
      />
    </Stack>
  );
};
