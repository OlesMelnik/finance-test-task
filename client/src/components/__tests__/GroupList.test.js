import { render } from "@testing-library/react";
import { GroupList } from "../GroupList";

test("matches snapshot", () => {
  const { baseElement } = render(<GroupList groups={[]} tickers={[]} />);
  expect(baseElement).toMatchSnapshot();
});
