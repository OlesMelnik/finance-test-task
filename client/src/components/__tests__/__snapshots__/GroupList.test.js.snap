// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<body>
  <div>
    <div
      class="css-e53awj-MuiStack-root"
    >
      <div
        class="css-1nkqw8n-MuiStack-root"
        data-testid="All"
      >
        <p>
          All
        </p>
        <p>
          0
        </p>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        style="text-transform: none;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        Новий список
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;
